configure-ironic.sh

```diff
@@ -1,38 +1,35 @@
 #!/usr/bin/bash
 
-# Get environment settings and update ironic.conf
-PROVISIONING_INTERFACE=${PROVISIONING_INTERFACE:-"provisioning"}
-IRONIC_IP=$(ip -4 address show dev "$PROVISIONING_INTERFACE" | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1)
-until [ ! -z "${IRONIC_IP}" ]; do
-  echo "Waiting for ${PROVISIONING_INTERFACE} interface to be configured"
-  sleep 1
-  IRONIC_IP=$(ip -4 address show dev "$PROVISIONING_INTERFACE" | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1)
-done
+. /bin/ironic-common.sh
+
 HTTP_PORT=${HTTP_PORT:-"80"}
 MARIADB_PASSWORD=${MARIADB_PASSWORD:-"change_me"}
 NUMPROC=$(cat /proc/cpuinfo  | grep "^processor" | wc -l)
 NUMWORKERS=$(( NUMPROC < 12 ? NUMPROC : 12 ))
 
+wait_for_interface_or_ip
+
 cp /etc/ironic/ironic.conf /etc/ironic/ironic.conf_orig
 
 crudini --merge /etc/ironic/ironic.conf <<EOF
 [DEFAULT]
 my_ip = $IRONIC_IP
 
 [api]
+host_ip = ::
```

Comment:
Required so host listens on 0.0.0.0 and IPv6

Reply:
Does this work if we set the provisioning network ipv6 address?

This is fine as a first step, but I think we should really bind to the provisioning nic ref https://github.com/metal3-io/ironic-image/pull/56

That was blocked originally because we needed to access both nics when running ironic on the host, but now that metal3-dev-env and dev-scripts run ironic in the cluster I think we could switch to just bind on PROVISIONING_IP ?

Reply:
It's required, the default otherwise is 0.0.0.0 so even if my_ip is IPv6, I Ironic won't respond

Reply:
Also, we could change host_ip to be the $IRONIC_IP, but that's not what it does today, Ironic listens on everything.

There's a few places we'd have to remove things like http://127.0.0.1 in some of the scripts to make that work. Which we probably should so it's locked down to only listening on the provisioning network.

Reply:
Ack, Ok lets go with this then I can revive #56 as a follow-up